### PR TITLE
fix: source status api response parsing for waiting for job

### PR DESCRIPTION
**Pull request by RanjeetMishra**
**Fixes**

- /sources/:jobRunId/status API response has been changed. Corresponding changes added in handling response.
- updated airflow because of critical and high vulnerabilities in older versions. This might cause issue for users using older version. They will have to upgrade Airflow version.

Test code updated and passes:
```
.[2023-04-10 13:41:43,196] {rudderstack.py:58} INFO - sync finished for sourceId: some-source-id
..[2023-04-10 13:41:43,201] {rudderstack.py:31} INFO - triggering sync for sourceId: some-source-id, endpoint: /v2/sources/some-source-id/start
.[2023-04-10 13:41:43,203] {rudderstack.py:31} INFO - triggering sync for sourceId: some-source-id, endpoint: /v2/sources/some-source-id/start
[2023-04-10 13:41:43,203] {rudderstack.py:36} INFO - Job triggered for sourceId: some-source-id
.[2023-04-10 13:41:43,204] {rudderstack.py:31} INFO - triggering sync for sourceId: some-source-id, endpoint: /v2/sources/some-source-id/start
[2023-04-10 13:41:43,204] {rudderstack.py:38} INFO - Job is already running for sourceId: some-source-id
.[2023-04-10 13:41:43,205] {rudderstack.py:31} INFO - triggering sync for sourceId: some-source-id, endpoint: /v2/sources/some-source-id/start
----------------------------------------------------------------------
Ran 7 tests in 60.838s
```

* Verified response for both ETL/RETL Jobs.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: rudder_airflow_provider/version.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '1.0.1'
\ No newline at end of file
+__version__ = '1.0.2'
```
Comment (RanjeetMishra) [inline rudder_airflow_provider/version.py:1]: Ideally this we should make 1.1.0 if we are upgrading airflow version required to higher.

File: rudder_airflow_provider/hooks/rudderstack.py [hunk 2]
```diff
@@ -47,11 +47,9 @@ def poll_for_status(self):
         while True:
             self.method = 'GET'
             resp = self.run(endpoint=status_endpoint, headers=headers).json()
-            job_status = resp['status']
-            logging.info('sync status for sourceId: %s, status: %s',
-                         self.source_id, job_status)
-
-            if job_status == STATUS_FINISHED:
+            finish_time = resp.get('finished_at') or None
```
Comment (am6010) [inline rudder_airflow_provider/hooks/rudderstack.py:50]:
```suggestion
            finish_time = resp.get(FINISHED_AS) or None
```
Minor change in line 7. 
`STATUS_FINISHED = 'finished' ->  FINISHED_AT= 'finished_at'`

**Review by am6010 — APPROVED** [review]

File: rudder_airflow_provider/hooks/rudderstack.py (hunk 2 above)
Comment (RanjeetMishra) [reply rudder_airflow_provider/hooks/rudderstack.py]: Fixed

File: rudder_airflow_provider/hooks/rudderstack.py [hunk 3]
```diff
@@ -47,11 +47,9 @@ def poll_for_status(self):
         while True:
             self.method = 'GET'
             resp = self.run(endpoint=status_endpoint, headers=headers).json()
-            job_status = resp['status']
-            logging.info('sync status for sourceId: %s, status: %s',
-                         self.source_id, job_status)
-
-            if job_status == STATUS_FINISHED:
+            finish_time = resp.get(FINISHED_AT) or None
```
Comment (vyeshwanth) [inline rudder_airflow_provider/hooks/rudderstack.py:50]: can we make this change in the API instead of provider?

Comment (vyeshwanth) [reply rudder_airflow_provider/hooks/rudderstack.py:50]: and `finished_at` will be '0001-01-01 00:00:00'

Comment (RanjeetMishra) [reply rudder_airflow_provider/hooks/rudderstack.py:51]: Fixed.

**Review by vyeshwanth — COMMENTED** [review]
@RanjeetMishra can we close this PR? we don't need these changes for tracking job status

**Pull request closed without merging**